### PR TITLE
Safety to avoid modifying files outside deploy_path

### PR DESCRIPTION
**Pull request by hansemschnokeloch**
Using writable_dirs with writable_use_sudo can lead to a disaster on a server (example using /var/log instead of var/log as writable dir).

- [ ] Bug fix #…?
- [ ] New feature?
- [ ] BC breaks?
- [ ] Deprecations?
- [ ] Tests added?
- [x] Changelog updated?

      Please, update CHANGELOG.md by running next command:
      $ php bin/changelog

### CODE REVIEW

**Comment by antonmedv** [comment]
Nice addition. 👍🏻

**Comment by martijngastkemper** [comment]
Maybe add this to the docs. Because this feature removes the possibility to use absolute paths.

**Comment by hansemschnokeloch** [comment]
I modified my draft. My first version was not safe enough, because with a space before the slash "  /var/log" the trim was not working anymore.
Now I add the deploy_path to the directories. I also added an option to disable this behavior and use absolute paths if needed.

File: recipe/deploy/writable.php [hunk 1]
```diff
@@ -39,12 +39,16 @@
 // For chmod mode only (if is boolean, it has priority over `writable_recursive`)
 set('writable_chmod_recursive', true);
 
+// Allow absolute path in writable_dirs
+set('writable_allow_absolute_path', false);
```
Comment (antonmedv) [inline recipe/deploy/writable.php:41]: I think it’s otherkill. I assume setting writable outside deploy path is really corner case and should be ignored.

**Comment by antonmedv** [comment]
But instead trimming slashes error message to user can be shown. It’s much better design.

**Comment by hansemschnokeloch** [comment]
I agree, throwing an error message is much better, and we need less code lines.

**Pull request merged**